### PR TITLE
Restrict access to the admin password file

### PR DESCRIPTION
**Pull request by CAFxX**
The admin password file was previously using the default owner/permissions of `/var/vcap/store`: `root:root` and `0644`. Given the sensitive nature of its contents, change to `root:root` and `0600`.

Also ensure that all changes have been persisted to disk at the end of the setup script (otherwise a host crash immediately after the execution of the setup script could have left the job in a broken state - e.g. if the admin_password_file hadn't been synced to disk).

Tested both upgrading from an existing deployment and creating a new deployment:
```
influxdb/e5667339-ba90-4e9c-ad86-982b1bd3ed51:~$ ls -al /var/vcap/store/influxdb/
total 24
drwxr-xr-x 5 vcap vcap 4096 Jun 20 04:20 .
drwxr-xr-x 4 root root 4096 Jun 20 04:20 ..
-rw------- 1 root root   33 Jun 20 04:20 admin_password_file
drwxr-xr-x 3 vcap vcap 4096 Jun 20 04:20 data
drwxr-xr-x 2 vcap vcap 4096 Jun 20 04:21 meta
drwx------ 3 vcap vcap 4096 Jun 20 04:20 wal
```

### CODE REVIEW

File: jobs/influxdb/templates/bin/influxdb_setup.erb [hunk 1]
```diff
@@ -71,10 +71,14 @@ INFLUXDB_HTTP_PORT=<%= p('influxdb.http.port') %>
 ADMIN_USERNAME="<%= admin_username %>"
 ADMIN_PASSWORD_FILE="/var/vcap/store/influxdb/admin_password_file"
 if [ ! -f "$ADMIN_PASSWORD_FILE" ]; then
-  # generate the admin password
+  # generate the admin password and store it in /var/vcap/store
   openssl rand -hex 16 > "$ADMIN_PASSWORD_FILE"
 fi
-ADMIN_PASSWORD=$(cat "$ADMIN_PASSWORD_FILE")
+if [ $(stat -c %U "$ADMIN_PASSWORD_FILE") != "root" ]; then # TODO: merge with the previous if
+  sudo chown root:root "$ADMIN_PASSWORD_FILE"
```
Comment (giner) [inline jobs/influxdb/templates/bin/influxdb_setup.erb]: What about moving this out to a new script and pass here as environment variable so that sudo is not required?

File: jobs/influxdb/templates/bin/influxdb_setup.erb [hunk 2]
```diff
@@ -71,10 +71,14 @@ INFLUXDB_HTTP_PORT=<%= p('influxdb.http.port') %>
...
+  sudo chmod 600 "$ADMIN_PASSWORD_FILE"
+fi
```
Comment (CAFxX) [inline jobs/influxdb/templates/bin/influxdb_setup.erb]: the separate `if` is required to migrate existing deployments. once all deployments have migrated the two `if`s can be merged.

**Review by XuejiaoZhang — APPROVED** [review]

**Review by b10s — APPROVED** [review]
LGTM

File: jobs/influxdb/templates/bin/influxdb_setup.erb [hunk 3]
```diff
@@ -72,11 +72,11 @@ ADMIN_USERNAME="<%= admin_username %>"
 ADMIN_PASSWORD_FILE="/var/vcap/store/influxdb/admin_password_file"
 if [ ! -f "$ADMIN_PASSWORD_FILE" ]; then
   # generate the admin password and store it in /var/vcap/store
-  openssl rand -hex 16 > "$ADMIN_PASSWORD_FILE"
+  xxd -l 16 -pu </dev/urandom >"$ADMIN_PASSWORD_FILE"
 fi
-sudo chmod 0600 "$ADMIN_PASSWORD_FILE"
-sudo chown root:root "$ADMIN_PASSWORD_FILE"
-ADMIN_PASSWORD=$(sudo cat "$ADMIN_PASSWORD_FILE")
+chmod 0600 "$ADMIN_PASSWORD_FILE"
```
Comment (b10s) [inline jobs/influxdb/templates/bin/influxdb_setup.erb]: why did you remove sudo?

File: jobs/influxdb/templates/bin/influxdb_setup.erb [hunk 4]
```diff
@@ -72,11 +72,11 @@ ADMIN_USERNAME="<%= admin_username %>"
 ADMIN_PASSWORD_FILE="/var/vcap/store/influxdb/admin_password_file"
 if [ ! -f "$ADMIN_PASSWORD_FILE" ]; then
   # generate the admin password and store it in /var/vcap/store
-  openssl rand -hex 16 > "$ADMIN_PASSWORD_FILE"
+  xxd -l 16 -pu </dev/urandom >"$ADMIN_PASSWORD_FILE"
```
Comment (b10s) [inline jobs/influxdb/templates/bin/influxdb_setup.erb]: Looks likes `u` here is useless:

```
$ xxd -l 16 -pu /dev/urandom
2770fa33e13d46973237ce69ea5beaa1
```

probably it was supposed to be:

```
$ xxd -u -l 16 -p /dev/urandom
C87E178B1E4448EF0053023A106F83D2
```

or without `u`

**Review by giner — APPROVED** [review]

File: jobs/influxdb/templates/bin/influxdb_ctl.erb [hunk 5]
```diff
@@ -19,12 +19,18 @@ exec 2> >( tee -a /var/vcap/sys/log/influxdb_ctl.err.log | logger -p user.error
 ulimit -c unlimited
 ulimit -n <%= p("influxdb.max_open_fd") %>
 
-for DIR in $DATA_DIR $STORE_DIR $LOG_DIR $RUN_DIR; do
```
Comment (b10s) [inline jobs/influxdb/templates/bin/influxdb_ctl.erb]: Looks like now it is going not only no `chown` but also no `mkdir` and no `chmod`.
Is it expected?

File: jobs/influxdb/templates/bin/influxdb_ctl.erb [hunk 6]
```diff
@@ -19,12 +19,18 @@ exec 2> >( tee -a /var/vcap/sys/log/influxdb_ctl.err.log | logger -p user.error
...
+for DIR in $DATA_DIR $LOG_DIR $RUN_DIR; do
```
Comment (b10s) [inline jobs/influxdb/templates/bin/influxdb_ctl.erb:22]: Why we do not need to `chown -R` `/var/vcap/store/influxdb`  ?

Comment (CAFxX) [reply jobs/influxdb/templates/bin/influxdb_ctl.erb:22]: because if you do it recursively it overwrites the owner/permissions of /var/vcap/store/influxdb/admin_password_file that as the PR description says, should be root:root/0600 (instead of vcap:vcap/755)
